### PR TITLE
3014 add new inbound return from list view

### LINKED ISSUE #3014: Add inbound return from the list view
## Is your feature request related to a problem? Please describe 👀

Need to be able to manually create inbound returns.

## Describe the solution you'd like 🎁

Create an empty inbound return from the inbound returns list view. 
Use the mock API insert mutation

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #3014 

# 👩🏻‍💻 What does this PR do? 
 

Selecting a customer here now calls the insertInboundReturn mutation and redirects to the detail URL (page incoming lol)
Screenshot 2024-02-21 at 1 43 12 PM

# 🧪 How has/should this change been tested? 

- On the inbound returns list view, click `New Return`
- Customer selection modal appears
- When you select a customer, the insert mutation is called, and you are redirected to the detail page

### CODE REVIEW

File: client/packages/common/src/intl/locales/en/distribution.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "button.new-shipment": "New Shipment",
   "button.select-a-color": "Select a colour",
   "button.stop": "Stop",
+  "error.failed-to-create-return": "Failed to create return!",
```
Comment (andreievg) [inline client/packages/common/src/intl/locales/en/distribution.json:14]: Wow just realised that we duplicate strings for distribution and replenishment

Comment (andreievg) [reply client/packages/common/src/intl/locales/en/distribution.json:14]: I've added a small comments here: https://github.com/msupply-foundation/open-msupply/issues/1266#issuecomment-1957995627

**Review by andreievg — APPROVED** [review]
Looks great, thanks

File: client/packages/common/src/intl/locales/en/distribution.json (hunk 1 above)
Comment (lache-melvin) [reply client/packages/common/src/intl/locales/en/distribution.json:14]: Ah awesome thanks Andrei, yeah would be great to consolidate those!

**Pull request merged**